optuna/study/_optimize.py

```diff
@@ -237,93 +222,37 @@ def _run_trial(
 
     # `Study.tell` may raise during trial post-processing.
     try:
-        study.tell(trial, values=values, state=state)
+        with warnings.catch_warnings(record=True) as tell_warnings:
```

Comment:
Do we have to enforce warnings with `simplefilter("always", ...)` in addition to this line?

Reply:
0d5a674 adds a filter.